### ISSUE TITLE
Support for absolute paths in `ignore` setting in `pyproject.toml`

### ISSUE THREAD

**Issue by simensol** [microsoft/pyright#9294]
While configuring the `ignore` setting in `pyproject.toml`, it appears that absolute paths are not supported. For example, the following configuration does not work:

```toml
[tool.pyright]
ignore = ["/usr/local/lib/python3.12/**"]
```

The log indicates that the path is not considered because it is not relative:
```
Ignoring path "/usr/local/lib/python3.12/**" in "ignore" array because it is not relative.
```

However, setting `ignore` via the VS Code `settings.json` or `devcontainer.json` does work with absolute paths:
```json
"python.analysis.ignore": ["/usr/local/lib/python3.12/**"]
```

This inconsistency suggests that absolute path support in `pyproject.toml` should be aligned with the behavior in `settings.json`/`devcontainer.json`. Alternatively, if absolute paths should not be supported, it would be helpful to document this explicitly and potentially remove support for absolute paths in `python.analysis.ignore` for consistency.

Reference: https://github.com/microsoft/pylance-release/issues/6044#issuecomment-2430034998

## Environment data

- Language Server version: `v2024.10.1`

**Comment by erictraut** [comment]
The "include", "exclude", "ignore", and "strict" settings in config files are designed to accept only relative paths. Config files are assumed to be checked in as part of a project, and these settings apply specifically to files in the project, so they should be relative to the project's root directory. That's the reason pyright specifically treats it as an error if you attempt to use an absolute path in this case. Is there a reason you want to specify absolute paths here?

I can see some use cases for supporting absolute paths in language server setting "python.analysis.ignore". Editor settings are often not checked in to a project and can differ from one developer to another. @debonte, you mentioned in the other thread that you see it as a problem that absolute paths are accepted in the case of the language server setting but not in the config file. I think this is by design. Do you buy this argument, or do you still think this inconsistency is a problem? If so, what would you propose we do to resolve the inconsistency? I'd prefer not to allow absolute paths in the config file because I think it will lead to abuse of the feature, but I think it's possible that some Pylance users are relying on the ability use absolute paths for "python.analysis.ignore".

**Comment by simensol** [comment]
> Is there a reason you want to specify absolute paths here?

Occasionally, Pylance reports errors within the devcontainer's Python path `/usr/local/lib/python3.12/`:

![Image](https://github.com/user-attachments/assets/38dbd916-8d5d-4bd5-9ba8-52dae79fef23)

![Image](https://github.com/user-attachments/assets/38b99c9a-53a7-48e8-8631-95cfe54b642e)

See also https://github.com/microsoft/pylance-release/issues/6044

Since the root of my project inside the devcontainer is `/workspaces/backend/`, I prefer to ignore `/usr/local/lib/python3.12/` over `../../usr/local/lib/python3.12/`. Since the project is built into a devcontainer, the absolute paths will be the same for all developers. However, I also acknowledge your point on relative paths, so I'm not sure what's best.

**Comment by debonte** [comment]
> you mentioned in the other thread that you see it as a problem that absolute paths are accepted in the case of the language server setting but not in the config file. I think this is by design. Do you buy this argument, or do you still think this inconsistency is a problem?

I think this is problematic when combined with the new "settings.json settings are ignored if a config file is present" behavior. Now if the user has a config file, they lose the ability to ignore absolute paths.

**Comment by erictraut** [comment]
> Occasionally, Pylance reports errors within the devcontainer's Python path

@debonte, does the Pylance team understand why this is happening? Does it occur only if the file is open in the editor, or do these errors persist once the file is closed? If they persist when the file is closed, then that seems like a real bug, and we should try to get to the bottom of it. I've experienced related behaviors for the past two months, and I have some evidence that it's a bug in VS Code where it's failing to send the "close file" LSP command.

**Comment by erictraut** [comment]
> I think this is problematic when combined with the new "settings.json settings are ignored if a config file is present" behavior.

Yeah, that makes sense. Maybe we should relax the "must be relative" requirement for the "ignore" setting specifically. I feel more strongly that absolute paths should be rejected for "include" and "exclude" settings because it doesn't make sense to allow project source files to live outside of the project directory hierarchy.

**Comment by debonte** [comment]
On another issue I suggested to @simensol that it was due to these files being opened in the editor, but he hasn't said one way or the other

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.386

**Comment by simensol** [comment]
> On another issue I suggested to @simensol that it was due to these files being opened in the editor, but he hasn't said one way or the other

Referring to my [screenshot above](https://github.com/microsoft/pyright/issues/9294#issuecomment-2430313673), are you asking if closing the file from my project that I'm currently working on, or closing `/usr/local/lib/python3.12/uuid.py`, removes the error? I have never opened the latter. In fact, `/usr/local/lib/python3.12/` is not even the selected Python interpreter in VS Code; it is `/workspaces/backend/.venv`.
